Modules/TPC/src/CheckOfPads.cxx

```diff
@@ -0,0 +1,377 @@
+      size_t end = moName.find("_2D");
+      auto histSubName = moName.substr(7, end - 7);
+      resultEV = Quality::Good;
+      resultMean = Quality::Good;
+
+      auto* canv = (TCanvas*)mo->getObject();
+      if (!canv) {
+        continue;
+      }
+      // Check all histograms in the canvas
+
+      mTotalMean = 0.;
+
+      for (int tpads = 1; tpads <= 72; tpads++) {
+        const auto padName = fmt::format("{:s}_{:d}", moName, tpads);
+        const auto histName = fmt::format("h_{:s}_ROC_{:02d}", histSubName, tpads - 1);
+        TPad* pad = (TPad*)canv->GetListOfPrimitives()->FindObject(padName.data());
+        if (!pad) {
+          mSectorsName.push_back("notitle");
+          mSectorsQuality_EV.push_back(Quality::Null);
+          continue;
+        }
+        TH2F* h = (TH2F*)pad->GetListOfPrimitives()->FindObject(histName.data());
+        if (!h) {
+          mSectorsName.push_back("notitle");
+          mSectorsQuality_EV.push_back(Quality::Null);
+          continue;
+        }
+        const std::string titleh = h->GetTitle();
+
+        mSectorsName.push_back(titleh);
+        // check if we are dealing with IROC or OROC
+        float totalPads = 0;
+        const int MaximumXBin = h->GetNbinsX();
+        const int MaximumYBin = h->GetNbinsY();
+        if (titleh.find("IROC") != std::string::npos) {
+          totalPads = 5280;
+        } else if (titleh.find("OROC") != std::string::npos) {
+          totalPads = 9280;
+        } else {
+          return Quality::Null;
+        }
+
+        float padSum = 0.;
+        int padsCount = 0;
+        float padStdev = 0.;
+        float padMean = 0.;
+
+        // Run twice to get the mean and the standard deviation
+        for (int runNo = 1; runNo <= 2; runNo++) {
+          // Run1: calculate single Pad total->Mean
+          // Run2: calculate standardDeviation from mean
+          for (int xBin = 1; xBin <= MaximumXBin; xBin++) {
+            for (int yBin = 1; yBin <= MaximumYBin; yBin++) {
+              const float binvalue = h->GetBinContent(xBin, yBin);
+              if (binvalue != 0) {
+                if (runNo == 1) {
+                  padSum += binvalue;
+                  padsCount++;
+                } else {
+                  padStdev += pow(binvalue - padMean, 2);
+                }
+              }
+            }
+          }
+
+          if (runNo == 1 && padSum > 0.) { // no need for div by 0 check, because if padSum > 0 -> padsCount>0
+            padMean = padSum / padsCount;
+          }
+        }
+
+        if (mEmptyCheck) {
+          if (padsCount > mMediumQualityLimit * totalPads) {
+            resultEmpty = Quality::Good;
+          } else if (padsCount < mBadQualityLimit * totalPads) {
+            resultEmpty = Quality::Bad;
+          } else {
+            resultEmpty = Quality::Medium;
+          }
+          mSectorsQuality_Empty.push_back(resultEmpty);
+          mEmptyPadPercent.push_back(1. - (float)padsCount / totalPads);
```

Comment:
totalPads can be `0` if title does not contain `IROC` or `OROC`. Either take care above and directly continue, or check for division by 0.